notebook/services/contents/fileio.py

```diff
@@ -100,7 +102,57 @@ def atomic_writing(path, text=True, encoding='utf-8', log=None, **kwargs):
         os.remove(tmp_path)
 
 
-class FileManagerMixin(object):
+
+@contextmanager
+def _simple_writing(path, text=True, encoding='utf-8', log=None, **kwargs):
+    """Context manager to write file without doing atomic writing
+    ( for weird filesystem eg: nfs).
+
+    Parameters
+    ----------
+    path : str
+      The target file to write to.
+
+    text : bool, optional
+      Whether to open the file in text mode (i.e. to write unicode). Default is
+      True.
+
+    encoding : str, optional
+      The encoding to use for files opened in text mode. Default is UTF-8.
+
+    **kwargs
+      Passed to :func:`io.open`.
+    """
+    # realpath doesn't work on Windows: http://bugs.python.org/issue9949
+    # Luckily, we only need to resolve the file itself being a symlink, not
+    # any of its directories, so this will suffice:
+    if os.path.islink(path):
+        path = os.path.join(os.path.dirname(path), os.readlink(path))
+
+    if text:
+        # Make sure that text files have Unix linefeeds by default
+        kwargs.setdefault('newline', '\n')
+        fileobj = io.open(path, 'w', encoding=encoding, **kwargs)
+    else:
+        fileobj = io.open(path, 'wb', **kwargs)
+
+    try:
+        yield fileobj
+    except:
+        fileobj.close()
+        raise
+
+    # Flush to disk
+    fileobj.flush()
+    if log:
+        log.debug( path + " saved using simple fs writing")
+    os.fsync(fileobj.fileno())
```

Comment:
The calls to flush and fsync in the atomic writing are an attempt to ensure that the file is genuinely written to disk (as opposed to stored in an OS cache) before we remove the previous copy of the file. With atomic writing disabled, I would take them out and just allow the OS to do whatever caching it wants to.